whelktool/scripts/cleanups/2020/04/lxl-3048.groovy

```diff
@@ -0,0 +1,32 @@
+PrintWriter scheduledForUpdating = getReportWriter("scheduled-updates")
+PrintWriter failedUpdating = getReportWriter("failed-updates")
+
+String where = "collection = 'bib' and deleted = false AND data#>>'\n" +
+        "{@graph,2,illustrativeContent}' LIKE '%code%'"
+
+selectBySqlWhere(where) { data ->
+    boolean changed = false
+
+    def (record, instance, work) = data.graph
+
+    if (work.illustrativeContent instanceof List) {
+        Iterator it = work.illustrativeContent.iterator()
+        while (it.hasNext()) {
+            Object entity = it.next()
+            if (!entity.containsKey("@id")) {
+                it.remove()
+                changed = true
+            }
+        }
+    } else if ( (!work.illustrativeContent instanceof Map) || !work.illustrativeContent.containsKey("@id")) {
```

Comment:
```suggestion
    } else if ( !(work.illustrativeContent instanceof Map) || !work.illustrativeContent.containsKey("@id")) {
```
because `!someMap == false`. ;) (This works anyway since if `work.illustrativeContent` ain't a List it's either a Map or *very* wrongly used (as in never) in practice.)

Reply:
Nice catch! Damn parentheses! :)